src/command_modules/azure-cli-botservice/azure/cli/command_modules/botservice/_help.py

```diff
@@ -0,0 +1,175 @@
+            --page-d myPageId --secret mySecret --token myToken
+"""
+helps['bot email create'] = """
+    type: command
+    short-summary: Create Email Channel on a Bot.
+    examples:
+        - name: Add Email Channel for a Bot
+          text: |-
+            az bot email create -n botName -g MyResourceGroup -a abc@outlook.com
+            -p password
+"""
+helps['bot msteams create'] = """
+    type: command
+    short-summary: Create MsTeams Channel on a Bot.
+    examples:
+        - name: Add MsTeams Channel for a Bot with calling enabled
+          text: |-
+            az bot msteams -n botName -g MyResourceGroup --enable-calling
+            --calling-web-hook https://www.myapp.com/
+"""
+helps['bot skype create'] = """
+    type: command
+    short-summary: Create Skype Channel on a Bot.
+    examples:
+        - name: Add Skype Channel for a Bot with messaging and screen sharing enabled
+          text: |-
+            az bot skype -n botName -g MyResourceGroup --enable-messaging
+            --enable-screen-sharing
+"""
+helps['bot kik create'] = """
+    type: command
+    short-summary: Create Kik Channel on a Bot.
+    examples:
+        - name: Add Kik Channel for a Bot.
+          text: |-
+            az bot kik create -n botName -g MyResourceGroup -u mykikname
+            -p password --key key --is-validated
+"""
+helps['bot directline create'] = """
+    type: command
+    short-summary: Create DirectLine Channel on a Bot with only v3 protocol enabled.
+    examples:
+        - name: Add DirectLine Channel for a Bot.
+          text: |-
+            az bot directline create -n botName -g MyResourceGroup --disablev1
+"""
+helps['bot telegram create'] = """
+    type: command
+    short-summary: Create Telegram Channel on a Bot.
+    examples:
+        - name: Add Telegram Channel for a Bot.
+          text: |-
+            az bot telegram create -n botName -g MyResourceGroup --access-token token
+            --is-validated
+"""
+helps['bot sms create'] = """
+    type: command
+    short-summary: Create Sms Channel on a Bot.
+    examples:
+        - name: Add Sms Channel for a Bot.
+          text: |-
+            az bot sms create -n botName -g MyResourceGroup --account-sid sid
+            --auth-token token --is-validated
+"""
+helps['bot slack create'] = """
+    type: command
+    short-summary: Create Slack Channel on a Bot.
+"""
+helps['bot authsetting'] = """
+    type: group
+    short-summary: Manage OAuth Connection Settings on a Bot.
+"""
+helps['bot authsetting create'] = """
+    type: command
+    short-summary: Create an OAuth Connection Setting on a Bot.
+    examples:
+        - name: Create a new OAuth Connection Setting on a Bot.
+          text: |-
+            az bot authsetting create -g MyResourceGroup -n botName -c myConnectionName
+            --client-id clientId --client-secret secret --scopes "scope1 scope2" --service google
```

Comment:
This should not be how scope is handled.  If it is, that's an issue.

`--scopes scope1 scope2`: Space-separated list
`--scopes "scope1" "scope2"`: Space-separated list
`--scopes "scope1 scope2":` A string with spaces

Reply:
in our case, scope is merely a string. it's actually parsed on the authentication provider service which is called by the botservice resource provider. the scope string may be space separated , comma separated or even colon separated depending on the service provider used (github, aadv2 etc). since it's just a string , my approach seems like the correct one. thoughts?

Reply:
No, it needs to be consistent with the rest of the CLI, which means you need to accept a space-separated list and which you can then join with spaces to send to the server. `' '.join(scopes)`

Reply:
but what about the case where the scope is a comma separated list rather than space separated? on our service side, we dont parse this string and rather pass it on to whatever service provider is requested (aadv2, github etc) .
i guess, the problem here is my naming of the parameter as "scopes" which means it should take a list? i can make it clearer by naming it as "--provider-scope-string". will that make it  consistent?

Reply:
You would accept a space-separated list and then concatenate that list into a string with either spaces or commas, whatever the service requires.

Reply:
the cli doesnt know what the scope separator is, since it's a pass-through property for us. only the user knows what separator to use. i can make it such that it's 2 parameters - scopes and scope separator. so something like --scopes foo bar --scope-separator "," . 
why not just accept a string with spaces/colons/other separator though?

Reply:
All of the bot service channels are strongly-typed, no? Then shouldn't you know which channels use which syntax?

We don't allow other forms of separating lists because it creates confusion for CLI customers. Space-separated is the standard in CLI 2.0 and so we are very particular about not letting patterns slip in from CLI 1.0 (which used comma-separated lists) because it just creates inconsistency and confusion. A scope-separator field might be an option, but it seems like there would be a known default for every channel type (Facebook uses comma, Teams uses colon, etc.)

Reply:
this isnt related to channels and is a completely new feature. Let me give an example to make this scenario clear - 
assume that a bot developer x has built a bot that lets you see and manage your `github` repositories. at the start of a conversation, the bot will require you to sign in into your `github`. however you (as a malignant user) can trick someone else (like me) by sending me this sign in link. if i signin, the bot will think it's your repos and now you can do anything you want with my github account.
this new feature helps protect against this scenario. the cli portion of it lets the bot developer register his github (or any other service provider) app credentials with botservice, so that we can secure the above transaction . previously this security would be on them to implement.
there are around 53 service providers that we support and each has its own fields and the way they implement scopes is different. This is why this information has not been captured in the swagger (also the bot developer knows what the separator would be ).
hope that makes it clear. 
given that space separated is the norm here, i think a scope-separator filed would satisfy both our requirements.

Reply:
I think your earlier suggestion of `--provider-scope-string` would be better.  We use `--scopes` elsewhere in the CLI and it is always the space-separated list variety. This would make it clear I think.